src/core/shape_inference/include/utils.hpp

```diff
@@ -44,37 +48,202 @@ void eltwise_shape_infer(const OpType* op, const std::vector<T>& input_shapes, s
 }
 
 namespace ov {
-namespace op {
+/**
+ * \brief Get the raw data as TResult object.
+ *
+ * \tparam T               TResult data type.
+ * \tparam TResult         Type of return object, must support creation of std::inserter. Default std::vector<T>.
+ * \tparam UnaryOperation  Unary function object applied on data with signature (T f(const U u)).
+ *
+ * \param et    Element type of input data.
+ * \param ptr   Pointer to data of type et.
+ * \param size  Data size as number of elements.
+ * \param func  Unary operation function object.
+ *
+ * \throws ov::AssertionFailure for not supported element type.
+ * \return Object of TResult with data from input pointer and transformed by unary operation.
+ */
+template <class T, class TResult = std::vector<T>, class UnaryOperation>
+TResult get_raw_data_as(const element::Type_t et, const void* const ptr, const size_t size, UnaryOperation&& func) {
+    TResult out;
+    auto out_it = std::inserter(out, out.end());
```

Comment:
```suggestion
    auto out_it = std::back_inserter(out);
```

Reply:
std::inserter selected as more containers support it.